test/test_sparse.py

```diff
@@ -1694,6 +1694,129 @@ def test_log1p(self):
             device=self.device)
         self._test_log1p_tensor(input, torch.zeros([5, 6, 0]))
 
+    def _test_neg_negative(self, sparse_tensor):
+        dense_tensor = sparse_tensor.to_dense()
+        expected_output = dense_tensor.neg()
+
+        for op in (torch.neg, torch.negative):
+            self.assertEqual(expected_output, op(sparse_tensor).to_dense())
+            sparse_tensor_out = torch.zeros_like(sparse_tensor)
+            op(sparse_tensor, out=sparse_tensor_out)
+            self.assertEqual(expected_output, sparse_tensor_out.to_dense())
+
+        self.assertEqual(expected_output, sparse_tensor.neg().to_dense())
```

Comment:
nit: these blocks could be combined with the current block by enumerating

```
for op in (torch.neg, torch.negative, torch.Tensor.neg, torch.Tensor.neg_, torch.Tensor.negative, torch.Tensor.negative_, operator.neg):
```

(You'd have to `import operator` somewhere for the last one). Then you could check 

```
if op in (torch.neg, torch.negative):
  # out checks
  ... 
```

And otherwise skip the out checks. To make handling the inplace ops easy you cloud always clone the input sparse tensor on every op. 

The current test is OK, however, if you don't want to bother.

Reply:
Thanks for the suggest @mruberry , I like this approach !